src/ui/components/Main/component.ts

```diff
@@ -240,6 +244,58 @@ export default class extends Component {
   toggleVisualizer() {
     this.isVisualizerShowing = !this.isVisualizerShowing;
   }
+
+  hideShare() {
+    this.shareIsVisible = false;
+
+    if(this.shareTrigger) {
+      setTimeout(function(trigger) {
+        trigger.focus();
+      }, 100, this.shareTrigger);
+    }
+
+    this.shareTrigger = null;
+  }
+
+  showShare(event) {
+    this.shareTrigger = event.target;
+    this.shareIsVisible = true;
+  }
+
+  download() {
+    this.jsZipIsLoading = true;
+
+    loadJSZip().then((JSZip) => {
+      this.jsZipIsLoading = false;
+      let zip = new JSZip();
```

Comment:
Not sure how to address this TypeScript warning:
/src/ui/components/Main/component.ts(256,17) Cannot use 'new' with an expression whose type lacks a call or construct signature.

This is from the JSZip library that is loaded dynamically.